### ISSUE TITLE
How to access elements of CNTK variables  in user defined loss functions, in Python API?

### ISSUE THREAD

**Issue by nikhilYadala** [Microsoft/CNTK#2040]
I am implementing a custom loss function, in which I need to access individual elements of the output of the Neural network ( coming from the final dense layer of the model defined) . The following is the example code..
`def custom_loss_function(z, nn_labels):`
      `                 ##### code  evaluating "pred" using CNTK.OPS package ############`
          `       ce = -(nn_labels*cntk.log(pred)+(1-nn_labels)*cntk.log(1-pred)) + (z[1]-0.3)*(z[1]-0.3)`
          `       return ce
`
`z` is the output of the model created ( a tensor of dim equal to size of final layer of network)
`nn_labels` is the labels variable mapped from ctf format input file read using cntk.io.streamDef

I have to pass this custom_loss_function as the loss parameter to create a `trainer` object of `cntk.Trainer`

1)Is it possible to retrieve the elements of z , nn_lables , for eg may be using:- ` z[1]`, `nn_labels[1] `etc ?.
How ever, `print( z[1])` doesn't print the first element of the vector , the std-out is `Composite(Dense): Input('Input3', [#, *], [1 x 1 x 300]) -> Output('classify', [#, *], [513])`

2),If it is possible, is the above implementation of `custom_loss_function` correct?? I doubt because,  when the `trainer.train_mini_batch(mini_batch_data)` is called,, the tensors `z`, and `nn_labels` get reshaped (due to addition of dynamic axes of length equal to size of mini_batch_data) .

Note: On running, the code doesn't give any error, converges, stops as well.

**Comment by ebarsoum** [comment]
Hi Nikhil,

Printing z[1] will not print the value, this is a symbolic graph, it get executed when you feed the data. Do print the actual value of z[1], you need to do something like z[1].eval({input_vat:[values]}). What z[1] element present? And can you show how you define your input variable?

Thanks,
Emad

**Comment by cha-zhang** [comment]
Close this due to lack of response. Feel free to reopen.
